### PR TITLE
Feature keycloak 10.0.1 - Keycloak component

### PR DESCRIPTION
**Pull request by canattofilipe**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
The access control mechanism is managed by Auth

* **What is the new behavior (if this is a feature change)?**
The access control mechanism is managed by Keycloak

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
The user and clientes should get an access token from Keycloak instead of Auth

* **Other information**:
Tickets: dojot/dojot#1531, dojot/dojot#1737

### CODE REVIEW

File: keycloak/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Keycloak
+
+Keycloak is an Open Source Identity and Access Management solution for modern Applications and Services, for more details about it vistit [Keycloak oficial repository](https://github.com/keycloak/keycloak)
```
Comment (thiagodpf) [inline keycloak/README.md:3]: vistit => visit

File: keycloak/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+## Keycloak inside dojot
+The Keycloak was adopt by dojot in order to centralize the access control and offer SSO (Single sing-on)
+
+## Keycloak and Kafka
+The dojot needs publish messages in Kafka after tenant creation and removal events, as this feature is not offered natively by Keycloak, the dojot team reach this target:
+
+- Adopting and customizing [keycloak-kafka](https://github.com/SnuK87/keycloak-kafka) SPI (Service Provider Interface)
+- Customizing Keycloak services module to trigger admin events for tenants's CRUD operations (It can be removed when newer versions of Keycloak implement it natively, to go further in this thread visit [KEYCLOAK-14313](https://issues.redhat.com/browse/KEYCLOAK-14313))
```
Comment (thiagodpf) [inline keycloak/README.md:12]: Very good!

File: keycloak/examples/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Docker-compose example
+
+This example makes available an enviroment with Keycloak integrated with Kafka.
+
+To run this example, type:
+```
+docker-compose up
+```
+
+To explore Keycloak GUI (graphical user interface) access http://localhost:8080 and login in with user ```admin``` and password ```admin```
```
Comment (thiagodpf) [inline keycloak/examples/README.md:10]: Suggestion:
_[...] and login in with [...]_ => _and log in with_

File: keycloak/README.md (hunk 1 above)
Comment (canattofilipe) [reply keycloak/README.md]: Thanks for the observation, it was fixed.

File: keycloak/examples/README.md (hunk 3 above)
Comment (canattofilipe) [reply keycloak/examples/README.md]: Thanks for the observation, it was fixed.

**Review by thiagodpf — DISMISSED** [review]

**Review by rascaraficci — COMMENTED** [review]
It looks very good. I just left few comments.

**Comment by giovannicuriel** [comment]
:eyes:

File: keycloak/examples/README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ ```
+ 
+Now check ```dojot-management.dojot.tenancy``` topic to verify if the messages was sent:
+```
+docker container exec -it dojot_keycloak_kafka_1 bash -c 'unset JMX_PORT && kafka-console-consumer.sh --bootstrap-server localhost:9092 --topic dojot-management.dojot.tenancy --from-beginning'
+```
+
+The following result is expected:
+```
+{"type":"CREATE","tenant":"myTenant"}
```
[the remark this answers is not in the document]
Comment (canattofilipe) [reply keycloak/examples/README.md:49]: Good point, to reach features like this its only evolve the [TenantMessage](https://github.com/canattofilipe/dojot/blob/feature-keycloak-10.0.1/keycloak/plugins/keycloak-kafka/src/main/java/com/github/snuk87/keycloak/kafka/beans/TenantMessage.java) object with the desired attribute and value.

**Review by thiagodpf — APPROVED** [review]

**Pull request merged**